discovery/gossiper_test.go

```diff
@@ -126,6 +128,12 @@ func (r *mockGraphSource) AddEdge(info *models.ChannelEdgeInfo,
 	r.mu.Lock()
 	defer r.mu.Unlock()
 
+	if r.addEdgeErrCode.IsSome() {
```

Comment:
I'm not sure that it's better to use that here because you'd still need to set an error variable and conditionally check for `nil` since `WhenSome` doesn't let you return from `AddEdge`?